src/events/broker.py

```diff
@@ -545,3 +652,7 @@ def update_peer_cluster_data(self) -> None:
         )
 
         # self.charm.on.config_changed.emit()  # ensure both broker+balancer get a changed event
+
+    def _on_peer_relation_departed(self, event: RelationDepartedEvent) -> None:
```

Comment:
**question:** Why `relation-departed` and not `relation-broken` here?

Reply:
I started with `relation-broken` but to my surprise, the hook never got fired :(

Reply:
Interesting! OK then, it's fine to keep.